libs/editor/egui_editor/src/editor.rs

```diff
@@ -1,46 +1,52 @@
-use crate::ast::Ast;
+use crate::appearance::{Appearance, Theme};
+use crate::ast::AST;
+use crate::buffer::Buffer;
 use crate::cursor::Cursor;
-use crate::cursor_types::DocByteOffset;
-use crate::debug_layer::DebugLayer;
-use crate::galley::GalleyInfo;
-use crate::layout_job::LayoutJobInfo;
-use crate::styled_chunk::StyledChunk;
+use crate::debug::DebugInfo;
+use crate::galleys::Galleys;
+use crate::layouts::LayoutJobInfo;
+use crate::styles::StyleInfo;
 use crate::test_input::TEST_MARKDOWN;
-use crate::theme::VisualAppearance;
+use crate::unicode_segs::UnicodeSegs;
+use crate::{ast, events, galleys, layouts, styles, unicode_segs};
 use egui::{Context, FontData, FontDefinitions, FontFamily, Ui, Vec2};
 use std::sync::Arc;
 
 pub struct Editor {
-    pub raw: String,
-
-    pub visual_appearance: VisualAppearance,
-
-    pub debug: DebugLayer,
-    pub text_unprocessed: bool,
-    pub cursor_unprocessed: bool,
```

Comment:
these fields removed (now local variables in `ui()` below), `initialized` field added for special handling on first frame (when galleys haven't been computed yet so events can't be handled), other fields just renamed/reordered